src/cargo-kani/src/args.rs

```diff
@@ -0,0 +1,196 @@
+    pub unwind: Option<u32>,
+    /// Turn on automatic loop unwinding
+    #[structopt(long)]
+    pub auto_unwind: bool,
+    /// Pass through directly to CBMC; must be the last flag
+    #[structopt(long, allow_hyphen_values = true)] // consumes everything
+    pub cbmc_args: Vec<OsString>,
+    /*
+    # Add flags that produce extra artifacts.
+    def add_artifact_flags(make_group, add_flag, config):
+        default_target = config["default-target"]
+        assert default_target is not None, \
+            f"Missing item in parser config: \"default-target\".\n" \
+            "This is a bug; please report this to https://github.com/model-checking/rmc/issues."
+
+        group = make_group(
+            "Artifact flags", "Produce artifacts in addition to a basic Kani report.")
+        add_flag(group, "--gen-c", default=False, action=BooleanOptionalAction,
+                 help="Generate C file equivalent to inputted program")
+        add_flag(group, "--gen-c-runnable", default=False, action=BooleanOptionalAction,
+                 help="Generate C file equivalent to inputted program; "
+                      "performs additional processing to produce valid C code "
+                      "at the cost of some readability")
+        add_flag(group, "--target-dir", type=pl.Path, default=default_target, metavar="DIR",
+                 help=f"Directory for all generated artifacts; defaults to \"{default_target}\"")
+
+    # Add flags needed for toggling and switching between outputs.
+    def add_output_flags(make_group, add_flag, config):
+
+        group = make_group("Output flags", "Toggle between different styles of output")
+        add_flag(
+            group,
+            "--output-format",
+            default=OutputStyle.OLD,
+            type=OutputStyle,
+            action=EnumAction,
+            help="Select the format for output")
+
+    # Add flags we don't expect end-users to use.
+    def add_developer_flags(make_group, add_flag, config):
+        group = make_group(
+            "Developer flags", "These are generally meant for use by Kani developers, and are not stable.")
+        add_flag(group, "--mangler", default="v0", choices=["v0", "legacy"],
+                 help="Change what mangler is used by the Rust compiler")
+        add_flag(group, "--use-abs", default=False, action=BooleanOptionalAction,
+                 help="Use abstractions for the standard library")
+        add_flag(group, "--abs-type", default="std", choices=["std", "kani", "c-ffi", "no-back"],
+                 help="Choose abstraction for modules of standard library if available")
+        add_flag(group, "--restrict-vtable", default=False, action=BooleanOptionalAction,
+                 help="Restrict the targets of virtual table function pointer calls")
+
+        */
+}
+
+#[derive(Debug, StructOpt)]
+pub struct CheckArgs {
+    // Rust argument parsers (/clap) don't have the convenient '--flag' and '--no-flag' boolean pairs, so approximate
```

Comment:
Do we need them?

Reply:
I am in favor of revisiting. This seems to be adding unnecessary complexity and it doesn't seem to be a common practice in rust anyway.